### DISCUSSION TITLE
Any way to load spectrogram data from server?

### DISCUSSION THREAD

**Discussion by melijimenez11** [katspaugh/wavesurfer.js#3979] [Q&A]
I'm working with huge audio files (7+ hours). My peaks are correctly loaded from a server to generate the audio wave, but Wavesurfer crashes my browser when generating the spectrogram. How can I load the spectrogram from a server the same way I load peaks for the audio wave? 

I'm a beginner to wavesurfer. I've looked at the docs but I'm confused. Would love to get it to work, as I'm taking full advantage of things like zoom and regions. I could generate the heatmap elsewhere and just put it as an image on my app, but I would lose access to those features (being able to navigate the heatmap alongside the audio wave. 

Detailed information or suggestions would be very welcome.

**Comment by katspaugh** [comment]
You can pass a data URL in the [`frequenciesDataUrl` option](https://github.com/katspaugh/wavesurfer.js/blob/main/src/plugins/spectrogram.ts#L331) which sadly isn't documented (yet).

**Reply by melijimenez11** [reply]
Thank you @katspaugh 

Do you know any tools that can be used to generate this JSON array on the back end? 

I'm having a hard time understanding the expected format.

**Reply by katspaugh** [reply]
I do not. Please circle back and document it if you find a solution.
